src/main/java/org/broadinstitute/consent/http/service/DataAccessRequestService.java

```diff
@@ -241,9 +232,30 @@ public DataAccessRequest insertDraftDataAccessRequest(User user, DataAccessReque
             now,
             dar.getData()
         );
+
+        syncDataAccessRequestDatasets(dar.getData().getDatasetIds(), dar.getReferenceId());
+
         return findByReferenceId(dar.getReferenceId());
     }
 
+    /**
+     * First delete any rows with the current reference id. This will allow us to keep (referenceId, dataset_id) unique
+     * Takes in a list of datasetIds and a referenceId and adds them to the dar_dataset collection
+     *
+     * @param datasetIds List of Integers that represent the datasetIds
+     * @param referenceId ReferenceId of the corresponding DAR
+     */
+    private void syncDataAccessRequestDatasets(List<Integer> datasetIds, String referenceId) {
+        List<DarDataset> darDatasets = datasetIds.stream()
+                .map(datasetId -> new DarDataset(referenceId, datasetId))
+                .collect(Collectors.toUnmodifiableList());
```

Comment:
Nice! I have seen wonkiness in the past with `toUnmodifiableList` ... if you find any, switch it out to a plain list.